diff --git a/library/x509write_csr.c b/library/x509write_csr.c
index c7c8032be..046cdee80 100644
--- a/library/x509write_csr.c
+++ b/library/x509write_csr.c
@@ -298,7 +298,7 @@ int mbedtls_x509write_csr_der( mbedtls_x509write_csr *ctx, unsigned char *buf,
     int ret;
     unsigned char *sig;
 
-    if( ( sig = mbedtls_calloc( 1, MBEDTLS_PK_SIGNATURE_MAX_SIZE ) ) == NULL )
+    if( ( sig = mbedtls_calloc( 1, MBEDTLS_PK_SIGNATURE_MAX_SIZE + 1) ) == NULL )
     {
         return( MBEDTLS_ERR_X509_ALLOC_FAILED );
     }
